include/ginkgo/core/base/lin_op.hpp

```diff
@@ -828,6 +828,39 @@ public:                                                                      \
                   "semi-colon warnings")
 
 
+/**
+ * Defines setters and getters for the parameters of a class, simplifying its
+ * construction by removing the repetitive typing of boiler plate code. The
+ * setter and getter functions will be named set_parameter_name(...) and
+ * get_parameter_name().
+ *
+ * The passed in parameter should be a member variable of the class and not the
+ * factory parameter. These setters and getters can be used to re-set the member
+ * variables after the Factory has been generated. Hence, the generate step need
+ * not be called after using the setter.
```

Comment:
In addition, I think we need to be careful about this. I think we should have a big warning maybe with the following example to show that this can be dangerous.

This is not true for a lot of cases: if you consider the BlockJacobi, if you put a setter for the block pointers or storage optimization, then you would definitely need to call the generate method again (although I believe we never should do that to begin with, that is just to show that this can be used dangerously).